apps/api/src/app/events/usecases/cancel-delayed/cancel-delayed.usecase.ts

```diff
@@ -7,20 +7,25 @@ export class CancelDelayed {
   constructor(private jobRepository: JobRepository) {}
 
   public async execute(command: CancelDelayedCommand): Promise<boolean> {
-    const job = await this.jobRepository.findOne({
-      _environmentId: command.environmentId,
-      transactionId: command.transactionId,
-      status: JobStatusEnum.DELAYED,
-    });
+    const jobs = await this.jobRepository.find(
+      {
+        _environmentId: command.environmentId,
+        transactionId: command.transactionId,
+        status: JobStatusEnum.DELAYED,
+      },
+      '_id'
+    );
```

Comment:
it will still find only the delay job - as its the only one that has DELAYED status. The others are pending. Or am I missing something?

Reply:
Never mind. Got you. its multiple subscribers